### PR TITLE
Tests: Refactoring of various things

### PR DESCRIPTION
**Pull request by StefanSpieker**
* fixed deprecation warnings, 
* simplified JUnit assertions and 
* removed obsolete throws Exception statements
* Added some missing `@NonNull`
* Use of StandardCharsets

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: test/src/test/java/hudson/LauncherTest.java [hunk 1]
```diff
@@ -271,10 +273,10 @@ private void assertMultipleStdioCalls(String message, Node node, boolean emitStd
         assertEquals(message, 0, ps.join());
         if (outputIn2) {
             assertThat(message, baos2.toString(), containsString("hello"));
-            assertThat(message, baos1.toString(), isEmptyString());
+            assertThat(message, baos1.toString(), is(emptyString()));
```
Comment (stellargo) [inline test/src/test/java/hudson/LauncherTest.java:276]: There are some * imports that can also be refactored, and `isEmptyString()` matcher can probably be not imported.

File: test/src/test/java/hudson/model/AbstractItemSecurityTest.java [hunk 2]
```diff
@@ -80,7 +82,7 @@ public void testUpdateByXmlDoesNotFail() throws Exception {
                 "</project>";
 
         FreeStyleProject project = jenkinsRule.createFreeStyleProject("security-167");
-        project.updateByXml((StreamSource) new StreamSource(new StringReader(xml)));
```
Comment (stellargo) [inline test/src/test/java/hudson/model/AbstractItemSecurityTest.java:83]: What would be the motivation behind this?

File: test/src/test/java/hudson/model/AbstractItemSecurityTest.java [hunk 3]
```diff
@@ -59,7 +61,7 @@ public void testUpdateByXmlDoesNotProcessForeignResources() throws Exception {
         FreeStyleProject project = jenkinsRule.createFreeStyleProject("security-167");
         project.setDescription("Wibble");
         try {
-            project.updateByXml(new StreamSource(new StringReader(xml)));
+            project.updateByXml((Source)new StreamSource(new StringReader(xml)));
```
Comment (stellargo) [inline test/src/test/java/hudson/model/AbstractItemSecurityTest.java:64]: What would be the motivation behind this?

File: test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java [hunk 4]
```diff
@@ -60,6 +62,7 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
```
Comment (stellargo) [inline test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java:63]: Why not remove this entirely? Also in the other edited files, and move completely to hamcrest.

File: test/src/test/java/hudson/model/ItemsTest.java [hunk 5]
```diff
@@ -242,7 +242,7 @@ private void cannotOverwrite(String target) throws Exception {
                         .withRedirectEnabled(false)
                         .withThrowExceptionOnFailingStatusCode(false);
                 WebResponse webResponse = wc.getPage(new WebRequest(new URL(wc.getContextPath() + "createItem?name=" + target + "&mode=hudson.model.FreeStyleProject"), HttpMethod.POST)).getWebResponse();
-                if (webResponse.getStatusCode() != HttpStatus.SC_MOVED_TEMPORARILY) {
+                if (webResponse.getStatusCode() != HttpURLConnection.HTTP_MOVED_TEMP) {
```
Comment (stellargo) [inline test/src/test/java/hudson/model/ItemsTest.java:245]: What would be the motivation behind this?

File: test/src/test/java/hudson/model/ParametersTest.java [hunk 6]
```diff
@@ -75,7 +76,7 @@ public void parameterTypes() throws Exception {
         Object o = DomNodeUtil.selectSingleNode(element, ".//input[@name='value']");
         System.out.println(o);
         HtmlCheckBoxInput booleanParameterInput = (HtmlCheckBoxInput) o;
-        assertEquals(true, booleanParameterInput.isChecked());
+        assertTrue(booleanParameterInput.isChecked());
```
Comment (stellargo) [inline test/src/test/java/hudson/model/ParametersTest.java:79]:
```suggestion
        assertThat(booleanParameterInput.isChecked(), is(true));
```
You can move this to hamcrest too like previous files.

File: test/src/test/java/hudson/security/HudsonPrivateSecurityRealmTest.java [hunk 7]
```diff
@@ -294,7 +294,7 @@ public void selfRegistrationTriggerLoggedIn() throws Exception {
 
         createAccountByAdmin("alice");
         // no new event in such case
-        assertEquals(true, spySecurityListener.loggedInUsernames.isEmpty());
+        assertTrue(spySecurityListener.loggedInUsernames.isEmpty());
```
Comment (stellargo) [inline test/src/test/java/hudson/security/HudsonPrivateSecurityRealmTest.java:297]: Here too hamcrest maybe?

File: test/src/test/java/hudson/tasks/FingerprinterTest.java [hunk 8]
```diff
@@ -259,8 +259,8 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         Collection<Fingerprint> fingerprints = action.getFingerprints().values();
         for (Fingerprint f: fingerprints) {
             assertTrue(f.getOriginal().is(upstream));
-            assertTrue(f.getOriginal().getName().equals(renamedProject1));
-            assertFalse(f.getOriginal().getName().equals(oldUpstreamName));
+            assertEquals(f.getOriginal().getName(), renamedProject1);
```
Comment (stellargo) [inline test/src/test/java/hudson/tasks/FingerprinterTest.java:262]: Here also use hamcrest?

**Comment by stellargo** [comment]
Great refactoring 👍 :)

File: test/src/test/java/hudson/model/ItemsTest.java (hunk 5 above)
Comment (StefanSpieker) [reply test/src/test/java/hudson/model/ItemsTest.java:245]: HttpStatus is deprecated, so I chose the newer version from java.net

File: test/src/test/java/hudson/model/AbstractItemSecurityTest.java (hunk 3 above)
Comment (StefanSpieker) [reply test/src/test/java/hudson/model/AbstractItemSecurityTest.java:64]: `public void updateByXml(StreamSource source)` is deprecated, and the call within this deprecated method is `updateByXml((Source)source);`

File: test/src/test/java/hudson/model/AbstractItemSecurityTest.java (hunk 2 above)
Comment (StefanSpieker) [reply test/src/test/java/hudson/model/AbstractItemSecurityTest.java:83]: see above.

File: test/src/test/java/hudson/model/ParametersTest.java (hunk 6 above)
Comment (StefanSpieker) [reply test/src/test/java/hudson/model/ParametersTest.java:79]: I really like the simplicity of assertTrue

**Review by oleg-nenashev — APPROVED** [review]
Looks good!

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Pull request merged**